hedera-node/hedera-token-service-impl/src/main/java/com/hedera/node/app/service/token/impl/handlers/transfer/AssociateTokenRecipientsStep.java

```diff
@@ -177,8 +180,35 @@ private TokenAssociation validateAndBuildAutoAssociation(
             validateTrue(validAssociations, NO_REMAINING_AUTOMATIC_ASSOCIATIONS);
             validateFalse(token.hasKycKey(), ACCOUNT_KYC_NOT_GRANTED_FOR_TOKEN);
             validateFalse(token.accountsFrozenByDefault(), ACCOUNT_FROZEN_FOR_TOKEN);
-            final var newRelation = autoAssociate(account, token, accountStore, tokenRelStore, config);
-            return asTokenAssociation(newRelation.tokenId(), newRelation.accountId());
+
+            final var unlimitedAssociationsEnabled =
+                    config.getConfigData(EntitiesConfig.class).unlimitedAutoAssociationsEnabled();
+            if (unlimitedAssociationsEnabled) {
+                final var syntheticAssociation = TransactionBody.newBuilder()
+                        .tokenAssociate(TokenAssociateTransactionBody.newBuilder()
+                                .account(account.accountId())
+                                .tokens(token.tokenId())
+                                .build())
+                        .build();
+                // We don't need to verify signatures for this internal dispatch. So we specify the keyVerifier to null
+                context.dispatchRemovablePrecedingTransaction(
+                        syntheticAssociation, SingleTransactionRecordBuilder.class, null, context.payer());
+                // increment the usedAutoAssociations count
+                final var accountModified = requireNonNull(accountStore.getAliasedAccountById(accountId))
+                        .copyBuilder()
+                        .usedAutoAssociations(account.usedAutoAssociations() + 1)
+                        .build();
+                accountStore.put(accountModified);
```

Comment:
I ran the e2e tests from the closed PR and some of them fail because they expect an automatic association count of 2 for 2 different tokens in a crypto transfer but the actual usedAutoAssociations is 0. Some of the old tests fail for the same assertion as well.

Reply:
One example with tests already in develop canMergeCreate2ChildWithHollowAccountHbarAndFungibleTransferUnlimitedAssociations

.hasAlreadyUsedAutomaticAssociations(1) assertions are failing when the flag is true in the test. I'm not sure which changes in this pr affect this though.

Reply:
@bilyana-gospodinova I have ported over most of the tests from your branch. Please have a look. Thanks

Reply:
@kselveliev  With the latest change. I see that test is passing locally. Could you please check

Reply:
The tests now pass. Thanks, @Neeharika-Sompalli.